File: app/local_plugins/ast_polychat/android/src/main/kotlin/com/aiagent/ast_polychat/AstPolychatPlugin.kt
package com.aiagent.ast_polychat

import io.flutter.embedding.engine.plugins.FlutterPlugin
import com.aiagent.plugin_interface.NativeServiceRegistry

/**
 * Flutter 插件入口：注册 vendor "polychat" 到 NativeServiceRegistry (AST)
 */
class AstPolychatPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val context = binding.applicationContext
        NativeServiceRegistry.registerAst("polychat") { AstPolychatService(context) }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: apps/client/local_plugins/vendors/ast_polychat/ios/Classes/AstPolychatPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// PolyChat AST Flutter plugin entry.
///
/// Mirrors the Android `AstPolychatPlugin`: registers vendor "polychat"
/// against `AstPolychatService`.
public class AstPolychatPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.ast_polychat", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        NativeServiceRegistry.shared.registerAst("polychat") { AstPolychatService() }
        os_log("Registered NativeAstService for vendor: polychat",
               log: log, type: .debug)
    }
}
